src/cloud_manager.ts

```diff
@@ -56,20 +55,53 @@ export default class CloudManager {
         groupCurrentCount: number,
         quantity: number,
         isScaleDownProtected: boolean,
-    ): Promise<boolean> {
+    ): Promise<number> {
         const groupName = group.name;
-        ctx.logger.info('Scaling up', { groupName, quantity });
+        ctx.logger.info('[CloudManager] Scaling up', { groupName, quantity });
         // TODO: get the instance manager by cloud
-        if (group.cloud == 'oracle') {
-            await this.oracleInstanceManager.launchInstances(
-                ctx,
-                group,
-                groupCurrentCount,
-                quantity,
-                isScaleDownProtected,
-            );
+        let scaleUpResult: Array<boolean | string>;
+
+        switch (group.cloud) {
+            case 'oracle':
+                scaleUpResult = await this.oracleInstanceManager.launchInstances(
+                    ctx,
+                    group,
+                    groupCurrentCount,
+                    quantity,
+                );
+                break;
+            default:
+                ctx.logger.error(`Cloud type not supported: ${group.cloud}`);
+                return 0;
         }
-        return true;
+
+        let scaleUpCount = 0;
+        scaleUpResult.forEach(async (instanceId) => {
```

Comment:
We might want to replace forEach with for/of loop. Initially I saw it as an IntelliJ suggestion then I read more and it doesn't seem appropriate to use await within a forEach() callback

Reply:
Agreed, fixed.